frontend/src/components/courses/forms/Edit.tsx

```diff
@@ -0,0 +1,182 @@
+import { Question, QuestionCategory } from "../../../models/courses/question";
+import { useParams } from "react-router-dom";
+import { CardStatus } from "../../utils/CardStatus";
+import { TablerColor } from "../../../utils/colors";
+import { QuestionEditor } from "./QuestionEditor";
+import { EditTooltipOption } from "./EditTooltip";
+
+export const EditCourseForm = () => {
+  const formId = parseInt(useParams<{ formId: string }>().formId);
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [tooltipIndex, setTooltipIndex] = useState<boolean | number>(false);
+  const newToast = useContext(ToastContext);
+
+  /* 
+    We don't want the questions to reload, as it could change our components!
+    This causes an issue because the components are loaded from a list, which we don't want to change
+    */
+  useEffect(
+    () => {
+      api.courses.forms.questions
+        .list(formId)
+        .then((questions) => {
+          setQuestions(questions);
+          setIsLoading(false);
+        })
+        .catch((err) => {
+          newToast.sendErrorToast("Could not fetch questions...");
+        });
+    },
+    // eslint-disable-next-line
+    []
+  );
+
+  const { data: form } = useBetterQuery<FormModel>(
+    ["courses.forms.get", formId],
+    api.courses.forms.get
+  );
+
+  const insertQuestion = (index = questions.length, duplicate = false) => {
+    let newQuestion: Question = {
+      label: "",
+      required: true,
+      archived: false,
+      category: QuestionCategory.Rating,
+      form: formId,
+    };
+    if (duplicate) {
+      newQuestion = Object.assign({}, questions[index]);
+    }
+    const copy = questions.slice();
+    copy.splice(index, 0, newQuestion);
+    setQuestions(copy);
+  };
+
+  const deleteQuestion = (index: number) => {
+    const copy = questions.slice();
+    copy.splice(index, 1);
+    setQuestions(copy);
+    setTooltipIndex(false);
+  };
+
+  const updateQuestion = (index: number, question: Question) => {
+    const copy = questions.slice();
+    copy.splice(index, 1, question);
+    setQuestions(copy);
+  };
+
+  const tooltipOptions: EditTooltipOption[] = [
+    {
+      icon: "plus",
+      onClick: (index) => insertQuestion(index, false),
+      tooltip: "ajouter",
+    },
+    {
+      icon: "copy",
+      onClick: (index) => insertQuestion(index, true),
+      tooltip: "copier",
+    },
+  ];
+
+  if (isLoading) return <p>Chargement des cours</p>;
+
+  return (
+    <Container>
+      <PageTitle>Modification de formulaire</PageTitle>
+
+      <Row>
+        <Col sm={11}>{form && <FormEditor form={form} />}</Col>
+      </Row>
+
+      <br />
+
+      {questions &&
+        questions.map((question, index) => {
+          const key: string = question.id + question.label;
+
+          return (
+            <Row key={"Row-" + key} onClick={() => setTooltipIndex(index)}>
```

Comment:
`key` is used only once, you might directly use it (with string interpolation) instead of separating declaration and usage